### PR TITLE
move Vec to vec:: and enhance documentation

### PR DESCRIPTION
**Pull request by BenjaminW3**
enhance documentation by:
* adding documentation
* moving Vec and Vec specific functions to `detail` and `vec` namespaces

### CODE REVIEW

**Comment by BenjaminW3** [comment]
push

File: test/unit/mem/buf/src/BufTest.cpp [hunk 1]
```diff
@@ -81,10 +81,11 @@ static auto createVecFromIndexedFn()
 -> alpaka::Vec<TDim, TSize>
 {
     return
+        alpaka::
```
Comment (psychocoderHPC) [inline test/unit/mem/buf/src/BufTest.cpp]: IMO this is very hard to read if the code is spitted by to many `#ifdef`

Could we change it back to the old style and only add `vec::` to the secound `#ifdef` part
```C++
  #ifdef ALPAKA_CREATE_VEC_IN_CLASS
         alpaka::Vec<TDim, TSize>::template
  #else
         vec::alpaka::
  #endif
```

File: test/unit/mem/view/src/ViewSubViewTest.cpp [hunk 2]
```diff
@@ -106,10 +106,11 @@ static auto createVecFromIndexedFn()
 -> alpaka::Vec<TDim, TSize>
 {
     return
+        alpaka::
```
Comment (psychocoderHPC) [inline test/unit/mem/view/src/ViewSubViewTest.cpp]: Could we change it back and add only `vec::` to the second if body

File: include/alpaka/vec/Vec.hpp [hunk 3]
```diff
@@ -988,7 +1014,9 @@ namespace alpaka
             using IdxOffset = std::integral_constant<std::intmax_t, static_cast<std::intmax_t>(dim::Dim<TExtent>::value) - static_cast<std::intmax_t>(TDim::value)>;
             return
 #ifdef ALPAKA_CREATE_VEC_IN_CLASS
-            Vec<TDim, size::Size<TExtent>>::template
+                Vec<TDim, size::Size<TExtent>>::template
```
Comment (psychocoderHPC) [inline include/alpaka/vec/Vec.hpp]: Could we move `alpaka::Vec<>` also to the namepsace `alpaka::vec`.
If we move `alpaka::Vec<>` to the namespace `alpaka::vec::` than we avoid a lot of the not so nice precompiler `#ifdef`'s

**Comment by psychocoderHPC** [comment]
Thanks for this pull request. I think we should move `alpaka::Vec` to the correct namespace to avoid to much `#ifdef`'s in the code.

File: include/alpaka/vec/Vec.hpp (hunk 3 above)
Comment (BenjaminW3) [inline include/alpaka/vec/Vec.hpp]: This would be my preferred solution but this will require changes to existing code using alpaka.

**Comment by BenjaminW3** [comment]
I will do it.

File: include/alpaka/vec/Vec.hpp (hunk 3 above)
Comment (psychocoderHPC) [inline include/alpaka/vec/Vec.hpp]: We should do this change now even if it need a change in the user code. This is only a change in the current dev which goes to the next release. We should add a no with interface changes to the release notes of the next release.

**Comment by BenjaminW3** [comment]
Done.

**Pull request merged**